### ISSUE TITLE
Creating an entity called Document when using mongodb leads to broken Application

### ISSUE THREAD

**Issue by The-Crocop** [jhipster/generator-jhipster#2515]
Basically the Problem is this. When you create an Entity called Document and use MongoDB at the same time the keyword Document is used twice in the same java file once like
import org.springframework.data.mongodb.core.mapping.Document;
and once for the class. which makes it not working. It shouldnt be allowed to create an entity called document in the first place and there should be a message with an error
Basically the error is: Document is already defined in this compilation unit

**Comment by jdubois** [comment]
Oh yes and the fix is easy: we should add this to our list of MongoDB reserved keywords.
I don't have my laptop with me, if you want to do a PR I will happily merge it.

**Comment by pascalgrimaud** [comment]
There are no list of MongoDB reserved keywords, so this list should be implemented.

**Comment by jdubois** [comment]
I can't believe we don't already have one!!! Yes this is needed.
